src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs

```diff
@@ -266,7 +266,7 @@ private bool IsMatchInBypassList(Uri input)
         /// </summary>
         public Uri GetProxy(Uri uri)
         {
-            return uri.Scheme == Uri.UriSchemeHttp ? _httpProxyUri : _httpsProxyUri;
+            return HttpUtilities.IsSupportedNonSecureScheme(uri.Scheme) ? _httpProxyUri : _httpsProxyUri;
```

Comment:
Hi @daddyman

we can use something like below instead of HttpUtilities.IsSupportedNonSecureScheme

uri.Scheme != Uri.UriSchemeWs ? _httpProxyUri : _httpsProxyUri

**Note** : We've to make below items public instead of internal as high lighted below in Uri.cs(codebase linked attached).

![image](https://user-images.githubusercontent.com/5323493/74760418-61f65500-52a0-11ea-84ed-d5865de465ba.png)

Code Base: https://github.com/dotnet/runtime/blob/91c1d7cab87653fc2214d9effbee54d3d9072c65/src/libraries/System.Private.Uri/src/System/Uri.cs

Happy to Pair with you on this :)

Reply:
Your suggestion doesn't work for http scheme. The check needs to be if the scheme is http or ws using the http proxy else use the https proxy. This is exactly what the HttpUtilities.IsSupportedNonSecureScheme method does.

I also think using HttpUtilities.IsSupportedNonSecureScheme is the best approach as it checks as the name of the method makes it clear what is being checked (answering the question of what is being asked in the if statement).

Changes to the Uri.cs are out of scope of this change as HttpUtilities class provides everything we need. I agree that it could be nice if the extra schemes were exposed by the Uri class but that should be a different issue/pull request.

Reply:
Thanks, mate for clarity ... you point make more clear why we have used HttpUtilities.IsSupportedNonSecureScheme